idx/cassandra/cassandra.go

```diff
@@ -216,48 +223,62 @@ func (c *CasIdx) Init() error {
 func (c *CasIdx) Stop() {
 	log.Info("cassandra-idx stopping")
 	c.MemoryIdx.Stop()
-	close(c.writeQueue)
+
+	// if updateCassIdx is disabled then writeQueue should never have been initialized
+	if updateCassIdx {
+		close(c.writeQueue)
+	}
 	c.wg.Wait()
 	c.session.Close()
 }
 
 func (c *CasIdx) AddOrUpdate(data *schema.MetricData, partition int32) {
 	pre := time.Now()
 	existing, inMemory := c.MemoryIdx.Get(data.Id)
+	updateIdx := false
+	stat := statUpdateDuration
 
 	if inMemory {
 		if existing.Partition == partition {
-			oldest := time.Now().Add(-1 * updateInterval).Add(-1 * time.Duration(rand.Int63n(updateInterval.Nanoseconds()*int64(updateFuzzyness*100)/100)))
+			var oldest time.Time
+			if updateInterval > 0 {
+				oldest = time.Now().Add(-1 * updateInterval).Add(-1 * time.Duration(rand.Int63n(updateInterval.Nanoseconds()*int64(updateFuzzyness*100)/100)))
+			} else {
+				oldest = time.Now()
+			}
 			if existing.LastUpdate < oldest.Unix() {
-				log.Debug("cassandra-idx def hasn't been seen for a while, updating index.")
-				def := schema.MetricDefinitionFromMetricData(data)
-				def.Partition = partition
-				c.MemoryIdx.AddOrUpdateDef(def)
-				c.writeQueue <- writeReq{recvTime: time.Now(), def: def}
-				statUpdateDuration.Value(time.Since(pre))
+				updateIdx = true
```

Comment:
we can replace 3 lines with one, if we do `updateIdx = (existing.LastUpdate < oldest.Unix())`